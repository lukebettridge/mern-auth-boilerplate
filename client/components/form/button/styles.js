import styled, { css } from "styled-components";

const Button = styled.button`
	background-color: #f8f8f8;
	border: 1px solid #eeeeee;
	cursor: pointer;
	font-size: 13pt;
	font-weight: 600;
	max-width: 350px;
	padding: 8px 20px;
	width: 100%;

	${props =>
		props.as === "a" &&
		css`
			background-color: transparent;
			color: #596f86;
			display: block;
			font-weight: 400;
			max-width: none;
			text-decoration: none;
			&:hover {
				text-decoration: underline;
			}

			${props.secondary &&
				css`
					color: #aaaaaa;
					font-size: 11pt;
					font-weight: 300;
				`}
		`}

	${props =>
		props.mt === "s" &&
		css`
			margin-top: 16px;
		`}

	${props =>
		props.mb === "s" &&
		css`
			margin-bottom: 16px;
		`}
`;

module.exports = { Button };
