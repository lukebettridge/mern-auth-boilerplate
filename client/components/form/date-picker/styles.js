import styled, { css } from "styled-components";

const DatePickerContainer = styled.div`
	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}

	${props =>
		(props.value || props.focused) &&
		css`
			& + label {
				font-size: 95%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}
`;

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Container = styled.div`
	position: relative;

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}

	.SingleDatePicker,
	.SingleDatePickerInput,
	.DateInput {
		display: block;
		width: 100%;
	}

	.DateInput_input {
		background-color: transparent;
		border-bottom: 1px solid #e9ecef;
		font-size: 11pt;
		font-weight: 300;
		letter-spacing: 1px;
		margin-bottom: 8px;
		padding: 12px 10px 5px;
		transition: border-color 100ms;
		width: 100%;

		&:hover {
			border-color: #b3b3b3;
		}

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}
	}
`;

const Label = styled.label`
	color: #999999;
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Container, DatePickerContainer, Error, Label };
