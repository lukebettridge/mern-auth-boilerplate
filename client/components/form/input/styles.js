import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Input = styled.input`
	background-color: transparent;
	border: 0;
	border-bottom: 1px solid #e9ecef;
	box-shadow: none;
	font-size: 11pt;
	font-weight: 300;
	margin-bottom: 8px;
	outline: none;
	padding: 12px 10px;
	width: 100%;

	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			border-color: #db1802;

			& + label {
				color: #db1802;
			}
		`}
`;

const InputContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 20px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Input, InputContainer, Label };
