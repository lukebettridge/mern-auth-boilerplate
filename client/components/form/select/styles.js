import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";
import { default as ReactSelect } from "react-select";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Select = styled(ReactSelect)`
	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		props.value.length > 0 &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}
        
	.react-select__control {
		appearance: none;
		background-color: transparent;
		border: 0;
		border-bottom: 1px solid #e9ecef;
		border-radius: 0;
		box-shadow: none;
		font-size: 12pt;
		font-weight: 300;
		height: 42px;
		margin-bottom: 8px;
		outline: none;
		width: 100%;

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}

		.react-select__input {
			opacity: 0;
		}
	}
`;

const SelectContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Select, SelectContainer, Label };
