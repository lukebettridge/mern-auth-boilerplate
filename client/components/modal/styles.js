import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

import ReactModal from "react-modal";

const Body = styled.div`
	@media (min-width: ${breakpoints.m}) {
		padding: 10px 20px;
	}

	${props =>
		!props.sideModal &&
		css`
			text-align: center;
		`}
`;

const Close = styled.a`
	height: 32px;
	opacity: 0.3;
	position: relative;
	width: 32px;

	&:hover {
		cursor: pointer;
		opacity: 1;
	}

	&:before,
	&:after {
		background-color: #333;
		content: " ";
		height: 33px;
		left: 15px;
		position: absolute;
		width: 2px;
	}

	&:before {
		transform: rotate(45deg);
	}

	&:after {
		transform: rotate(-45deg);
	}
`;

const Header = styled.div`
	display: flex;
	flex-direction: row-reverse;
	justify-content: space-between;
	margin-bottom: 30px;

	@media (min-width: ${breakpoints.m}) {
		padding-left: 20px;
	}
`;

const Heading = styled.h2`
	font-weight: 700;
	margin: 0;
`;

const Modal = styled(ReactModal)`
	background-color: #ffffff;
	border: 1px solid #e9ecef;
	bottom: auto;
	left: 50%;
	max-width: 650px;
	outline: none;
	padding: 20px 20px 62px;
	position: absolute;
	right: auto;
	top: 43%;
	width: calc(100% - 20px);

	${props =>
		props.sideModal
			? css`
					border: 0;
					border-left: 1px solid #e9ecef;
					bottom: 0;
					left: auto;
					max-width: 720px;
					overflow-y: auto;
					padding: 40px 20px;
					right: -100%;
					top: 0;
					transition: right 0.5s;
					width: 100%;
					z-index: 2;

					&.ReactModal__Content--after-open {
						right: 0;
					}

					@media (min-width: ${breakpoints.m}) {
						padding: 60px;
					}
			  `
			: css`
					border-radius: 2px;
					padding-bottom: 62px;
					transform: translate(-50%, -50%);
			  `}
`;

module.exports = { Body, Close, Header, Heading, Modal };
