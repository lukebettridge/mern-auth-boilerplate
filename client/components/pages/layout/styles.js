import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

import { Link as RouterLink } from "react-router-dom";

const anchorStyles = `
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}
`;

const subAnchorStyles = `
	${anchorStyles}
	font-size: 11pt;
	padding-left: 15px;
`;

const Anchor = styled.a`
	${anchorStyles}
	${props =>
		props.active &&
		css`
			opacity: 1;
		`}
`;

const Container = styled.div`
	display: flex;
	height: 100%;
`;

const Link = styled(RouterLink)`
	${anchorStyles}
`;

const List = styled.ul`
	margin: 0;
`;

const ListItem = styled.li`
	margin-bottom: 40px;
`;

const Slider = styled.div`
	align-items: center;
	background-color: #fbfbfc;
	border-left: 1px solid #e9ecef;
	border-bottom: 1px solid #e9ecef;
	color: #294661;
	display: flex;
	height: 50px;
	justify-content: center;
	position: absolute;
	right: 0;
	top: 0;
	width: 50px;
	z-index: 1;
	@media (min-width: ${breakpoints.m}) {
		display: none;
	}
`;

const SubAnchor = styled.a`
	${subAnchorStyles}
`;

const SubLink = styled(RouterLink)`
	${subAnchorStyles}
`;

const SubList = styled.ul`
	margin-top: 15px;
	max-height: 1000px;
	overflow: hidden;
	transition: all 0.3s;

	${props =>
		!props.active &&
		css`
			margin: 0;
			max-height: 0;
		`}
`;

const SubListItem = styled.li`
	margin-bottom: 8px;
`;

const Navigation = styled.div`
	background-color: #fbfbfc;
	border-right: 1px solid #e9ecef;
	display: none;
	height: 100%;
	max-width: 200px;
	min-width: 200px;
	padding-left: 15px;
	padding-right: 10px;
	padding-top: 48px;
	width: 200px;
	z-index: 1;

	${props =>
		props.active &&
		css`
			display: table-cell;
			position: absolute;
		`}

	@media (min-width: ${breakpoints.m}) {
		display: table-cell;
		height: unset;
		position: unset;
	}
`;

module.exports = {
	Anchor,
	Container,
	Link,
	List,
	ListItem,
	Navigation,
	Slider,
	SubAnchor,
	SubLink,
	SubList,
	SubListItem
};
