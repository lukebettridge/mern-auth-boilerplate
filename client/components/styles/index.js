/* eslint-disable indent */
import styled, { css } from "styled-components";
import breakpoints from "./breakpoints";

import { Link as ReactRouterLink } from "react-router-dom";

const Box = styled.div`
	background-color: #ffffff;
	border: 1px solid #e9ecef;
	color: #666666;
	margin-bottom: 20px;
	padding: 40px 30px;
	text-align: center;

	@media (min-width: ${breakpoints.m}) {
		padding: 54px;
		padding-bottom: 50px;
	}

	${props =>
		props.grey &&
		css`
			background-color: #e8e8e8;
			border-color: #c3c3c3;

			hr {
				border-color: #e0e0e0;
			}
		`}

		${props =>
			props.border &&
			css`
				border: ${props.border};
			`}

	${props =>
		props.mb === "none" &&
		css`
			margin-bottom: 0;
		`}

	${props =>
		props.padding === "none" &&
		css`
			padding: 0 !important;
		`}
`;

const Container = styled.div`
	margin: 0 auto;
	padding: 32px 15px;
	width: 100vw;

	@media (min-width: ${breakpoints.m}) {
		max-width: calc(100vw - 200px);
		padding: 32px 48px;
	}

	${props =>
		props.small &&
		css`
			max-width: 540px;
			@media (min-width: ${breakpoints.m}) {
				max-width: 700px;
			}
		`}

	${props =>
		props.tableCell &&
		css`
			display: table-cell;
		`}

		${props =>
			props.inactive &&
			css`
				opacity: 0.3;

				@media (min-width: ${breakpoints.m}) {
					opacity: 1;
				}
			`}
`;

const FilterWrap = styled.div`
	border: 1px solid #e9ecef;
	margin-bottom: 30px;
`;

const FilterHeader = styled.div`
	background-color: #f4f6f7;
	color: #546b81;
	border-bottom: 1px solid #e9ecef;
	display: flex;
	justify-content: space-between;
	padding: 10px 20px;
`;

const FilterBody = styled.div`
	background-color: #ffffff;
	display: flex;
	padding: 20px;
`;

const Heading = styled.h1`
	color: #212529;
	font-size: 30pt;
	font-weight: 900;
	letter-spacing: -1px;
	margin-top: 0;
	margin-bottom: 32px;
`;

const linkStyles = props => `
	color: #489be8;
	cursor: pointer;
	display: block;
	font-size: 11pt;
	font-weight: 300;
	margin: 10px;
	text-decoration: none;
	transition: opacity 0.3s;

	${
		props.secondary
			? css`
					color: #888888;
					opacity: 0.6;

					&:hover {
						opacity: 1;
					}
			  `
			: ""
	}

	${
		props.mt === "l"
			? css`
					margin-top: 35px;
			  `
			: ""
	}
	
`;

const Link = styled.a`
	${props => linkStyles(props)}
`;

const Paragraph = styled.p`
	font-weight: 300;
	line-height: 1.9;
	margin-top: 0;
	margin-bottom: 16px;

	${props =>
		props.light &&
		css`
			color: #cccccc;
			font-size: 10pt;
		`}

	${props =>
		props.center &&
		css`
			text-align: center;
		`}

	${props =>
		props.mt === "l" &&
		css`
			margin-top: 35px;
		`}

	${props =>
		props.mb === "l" &&
		css`
			margin-bottom: 35px;
		`}

	a {
		color: inherit;
		text-decoration: none;
		&:hover {
			text-decoration underline;
		}
	}
`;

const RouterLink = styled(ReactRouterLink)`
	${props => linkStyles(props)}
`;

const Table = styled.table`
	border-collapse: collapse;
	border-spacing: 0;
	color: #444444;
	width: 100%;

	thead {
		tr {
			background-color: transparent;
			border-color: transparent;

			th {
				color: #7f90a0;
				font-size: 9pt;
				letter-spacing: 1px;
				line-height: 15px;
				text-align: left;
				text-transform: uppercase;
			}
		}
	}

	tbody {
		tr {
			background-color: #ffffff;
			border: 1px solid #e9ecef;
			cursor: pointer;
			transition: background-color 0.3s;

			&:hover {
				background-color: #f4f6f7;
			}

			th {
				color: #546b81;
				text-align: left;
				text-transform: uppercase;
			}

			td {
				background-color: inherit;
			}
		}
	}

	tr {
		border-bottom: 0;
		font-size: 13px;
		line-height: 20px;
	}

	th,
	td {
		border: 0;
		padding: 20px;
		vertical-align: middle;

		${() =>
			[...Array(5)]
				.map((_, i) => `&.col-${i + 1} { grid-column: span ${i + 1}; }`)
				.join(" ")}
	}
`;

const TableAction = styled.a`
	background-color: #fbfbfc;
	border: 1px solid #e9ecef;
	color: #489be8;
	border-top: 0;
	display: block;
	font-size: 11pt;
	margin-bottom: 10px;
	padding: 15px;
	text-align: center;
`;

const Status = styled.span`
	display: inline-block;
	line-height: 20px;
	padding-left: 24px;
	position: relative;

	&:before {
		background-color: #d4dadf;
		border-radius: 50%;
		content: "";
		height: 12px;
		left: 0;
		position: absolute;
		top: 4px;
		width: 12px;

		${props =>
			props.success &&
			css`
				background-color: #18c96e;
			`}
	}
`;

const Subheading = styled.h2`
	font-size: 24pt;
	font-weight: 700;
	margin-top: 0;
	margin-bottom: 30px;
`;

export {
	Box,
	Container,
	FilterBody,
	FilterHeader,
	FilterWrap,
	Heading,
	Link,
	Paragraph,
	RouterLink,
	Table,
	TableAction,
	Status,
	Subheading
};
