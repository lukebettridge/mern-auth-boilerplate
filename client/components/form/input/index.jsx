import React, {
	forwardRef,
	useEffect,
	useImperativeHandle,
	useState
} from "react";
import PropTypes from "prop-types";

import * as utils from "./utils";
import * as S from "./styles";

const Input = forwardRef((props, ref) => {
	const [state, setState] = useState({
		error: props.error || "",
		value: props.value || ""
	});

	useEffect(() => {
		const { error } = props;
		setState(prev => ({ ...prev, error }));
	}, [props.error]);

	useImperativeHandle(ref, () => ({ validate, value: state.value }));

	const onBlur = e => {
		validate(true);
		if (props.onBlur) props.onBlur(e);
	};

	const onChange = e => {
		const value = e.target ? e.target.value : state.value;
		setState(prev => ({ ...prev, error: "", value }));
		if (props.onChange) props.onChange(e);
	};

	const validate = (onBlur = false) => {
		const error = utils.validate(props);
		setState(prev => ({
			...prev,
			error: !onBlur ? props.error || error : error
		}));
		return error;
	};

	const id = props.id || props.name;

	return (
		<S.InputContainer className="inputContainer" {...props}>
			<S.Input
				{...props}
				id={id}
				inError={!!state.error}
				onBlur={onBlur}
				onChange={onChange}
				ref={ref}
				value={state.value}
			/>
			{props.label && <S.Label htmlFor={id}>{props.label}</S.Label>}
			{!!state.error && <S.Error>{state.error}</S.Error>}
		</S.InputContainer>
	);
});

Input.propTypes = {
	error: PropTypes.string,
	friendlyName: PropTypes.string,
	id: PropTypes.string,
	isRequired: PropTypes.bool,
	label: PropTypes.string,
	max: PropTypes.number,
	min: PropTypes.number,
	name: PropTypes.string,
	onBlur: PropTypes.func,
	onChange: PropTypes.func,
	pattern: PropTypes.any,
	placeholder: PropTypes.string,
	type: PropTypes.string,
	value: PropTypes.any
};

Input.defaultProps = {
	placeholder: "",
	type: "text"
};

export default Input;
