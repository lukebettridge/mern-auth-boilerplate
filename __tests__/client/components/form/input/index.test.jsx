import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import * as S from "components/form/input/styles";
import Input from "components/form/input";

describe("Input component", () => {
	it("snapshot renders", () => {
		const subject = shallow(<Input />);

		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("renders label", () => {
		const subject = shallow(<Input label="foo" />);

		expect(subject.find(S.Label).length).toEqual(1);
		expect(subject.find(S.Label).text()).toEqual("foo");
	});

	it("renders error", () => {
		const subject = shallow(<Input error="foo" />);

		expect(subject.find(S.Error).length).toEqual(1);
		expect(subject.find(S.Error).text()).toEqual("foo");
	});

	it("calls onBlur prop", () => {
		const onBlurMock = jest.fn();
		const subject = shallow(<Input onBlur={onBlurMock} />);

		subject.find(S.Input).simulate("blur", "foo");

		expect(onBlurMock).toHaveBeenCalledWith("foo");
	});

	it("does not call onBlur prop", () => {
		const subject = shallow(<Input />);

		subject.find(S.Input).simulate("blur", "foo");

		expect(subject.props().onBlur).toBeUndefined();
	});

	it("calls onChange prop", () => {
		const onChangeMock = jest.fn();
		const subject = shallow(<Input onChange={onChangeMock} />);

		subject.find(S.Input).simulate("change", "foo");

		expect(onChangeMock).toHaveBeenCalledWith("foo");
	});

	it("does not call onChange prop", () => {
		const subject = shallow(<Input />);

		subject.find(S.Input).simulate("change", "foo");

		expect(subject.props().onChange).toBeUndefined();
	});

	it("calls onFocus prop", () => {
		const onFocusMock = jest.fn();
		const subject = shallow(<Input onFocus={onFocusMock} />);

		subject.find(S.Input).simulate("focus", "foo");

		expect(onFocusMock).toHaveBeenCalledWith("foo");
	});

	it("does not call onFocus prop", () => {
		const subject = shallow(<Input />);

		subject.find(S.Input).simulate("focus", "foo");

		expect(subject.props().onFocus).toBeUndefined();
	});

	it("sets placeholder", () => {
		const subject = shallow(<Input placeholder="foo" />);
		expect(subject.find(S.Input).props().placeholder).toEqual("foo");
	});

	it("sets placeholder with label", () => {
		const subject = shallow(<Input label="bar" placeholder="foo" />);
		expect(subject.find(S.Input).props().placeholder).toEqual("");
	});

	it("sets placeholder with label on focus", () => {
		const subject = shallow(<Input label="bar" placeholder="foo" />);

		subject.find(S.Input).simulate("focus", "foo");

		expect(subject.find(S.Input).props().placeholder).toEqual("foo");
	});
});
