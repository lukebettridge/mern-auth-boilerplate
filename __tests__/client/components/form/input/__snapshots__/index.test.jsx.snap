// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component snapshot renders 1`] = `
<styles__InputContainer
  className="inputContainer"
  placeholder=""
  type="text"
>
  <styles__Input
    inError={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</styles__InputContainer>
`;
